build/templates/setup.py.mako

```diff
@@ -46,7 +46,7 @@ setup(
     extras_require={
         'grpc': [
             'grpcio>=1.59.0,<2.0',
-            'protobuf>=4.21.6,<5.0'
+            'protobuf>=4.21.6,<6.0'
```

Comment:
Based on the 4.x -> 5.x upgrade, I think we should remove the `protobuf` upper bound entirely.

Thinking back to 3.x -> 4.x: it was possible to use the old 3.x codegen with 4.x, but I think you had to set an environment variable.

Reply:
Hmm, yeah, I suppose it's probably safe enough to remove.
If there's no wheel for a later version, it'll pick an older one.
If it installs but is somehow incompatible, the user can add their own installation constraints, but they can't overcome an upper bound that we include, if they need a later version.

Reply:
Upper bound removed.